src/middleware/node/get-payload.ts

```diff
@@ -11,18 +11,30 @@ import AggregateError from "aggregate-error";
 // }
 type IncomingMessage = any;
 
-export function getPayload(request: IncomingMessage): Promise<string> {
+
+async function buffer(readable: Readable) {
+  const chunks = [];
+  for await (const chunk of readable) {
+    chunks.push(typeof chunk === 'string' ? Buffer.from(chunk) : chunk);
+  }
+  return Buffer.concat(chunks);
+}
+
+export async function getPayload(request: IncomingMessage): Promise<string> {
   if ("body" in request) {
     if (
       typeof request.body === "object" &&
       "rawBody" in request &&
       request.rawBody instanceof Buffer
     ) {
       // The body is already an Object and rawBody is a Buffer (e.g. GCF)
-      return Promise.resolve(request.rawBody.toString("utf8"));
+      return request.rawBody.toString("utf8");
+    } else if(Symbol.asyncIterator in request || Symbol.iterator in request) {
```

Comment:
```suggestion
    // request is a readable stream of buffers (e.g. vercel)
    } else if(Symbol.asyncIterator in request || Symbol.iterator in request) {
```

Is it enough to check for the Symbols to assume it is a readable stream?

Reply:
This can handle some other cases where request can be an iterator over either strings or buffers. Whatever is behind it does not really matter.

Reply:
> This can handle some other cases where request can be an iterator over either strings or buffers

What other cases? This is the first time I know of that this has been raised, and this is a lot more work than just checking for `rawBody`.